apps/docs-snippets2/src/wallets/wallet-optional-provider.ts

```diff
@@ -0,0 +1,18 @@
+// #region wallet-optional-provider
+import type { WalletUnlocked } from 'fuels';
+import { Wallet } from 'fuels';
+
+// You can create a wallet, without a provider
+let unlockedWalletWithoutProvider: WalletUnlocked = Wallet.generate();
+unlockedWalletWithoutProvider = Wallet.fromPrivateKey(
+  unlockedWalletWithoutProvider.privateKey
+);
+
+// All non-provider dependent methods are available
+unlockedWalletWithoutProvider.lock();
+
+// All provider dependent methods will throw
+await expect(() => unlockedWalletWithoutProvider.getCoins()).rejects.toThrow(
+  /Provider not set/
+);
```

Comment:
How is `expect` being imported here? And should it be outside the snippet?

Reply:
I think `expect` is a ubiquitous BDD keyword to be used in the snippet and also I'm not sure how else I could articulate that the wallet needs a provider (other than it will throw an error if a provider method is called)

Reply:
You can use a `try/catch` and console.log the error:
```ts
// All provider dependent methods will throw
try { 
  await unlockedWalletWithoutProvider.getCoins() 
} catch(e) { 
  console.log('error', e) 
}
```

Even though `expect` does its thing here (and is more descriptive, tbh), one of the guiding principles we set with this endeavor was copy-pastable snippets, and having the `expect` here doesn't fulfill that, while a `try/catch` is good enough in this case that we don't have to think about alternatives.

Reply:
I concede that's a valid point @nedsalk in terms of copy/pastability, although I do think it's less elegant. 

6d5cc59ca